File: java/src/main/java/com/example/ddd/application/UserQueryHandler.java
package com.example.ddd.application;

import java.util.List;

/**
 * 用户查询Handler接口
 */
public interface UserQueryHandler {
    /**
     * 根据id查询，找不到返回null
     */
    UserDto queryById(String id);

    /**
     * 根据id查询，找不到抛出异常
     */
    UserDto queryByIdReq(String id);

    /**
     * 查询记录数
     */
    int queryCount(UserQueryCriteria criteria);

    /**
     * 查询
     */
    List<UserDto> query(UserQueryCriteria criteria, UserQuerySort... sorts);

    /**
     * 分页查询
     */
    List<UserDto> queryByPage(int pageNum, int pageSize, UserQueryCriteria criteria, UserQuerySort... sorts);
}

File: backend-example-python/src/application/user_query_handler.py
from abc import ABC, abstractmethod

from src.application.user_dto import UserDto
from src.application.user_query_criteria import UserQueryCriteria
from src.application.user_query_sort import UserQuerySort


class UserQueryHandler(ABC):
    """用户查询Handler接口"""

    @abstractmethod
    async def query_by_id(self, id: str) -> UserDto | None:
        """根据 id 查询，找不到返回 None"""
        pass

    @abstractmethod
    async def query_by_id_req(self, id: str) -> UserDto:
        """根据 id 查询，找不到抛出异常"""
        pass

    @abstractmethod
    async def query_count(self, criteria: UserQueryCriteria) -> int:
        """查询记录数"""
        pass

    @abstractmethod
    async def query(self, criteria: UserQueryCriteria, *sorts: UserQuerySort) -> list[UserDto]:
        """查询"""
        pass

    @abstractmethod
    async def query_by_page(self, page_num: int, page_size: int, criteria: UserQueryCriteria, *sorts: UserQuerySort)\
            -> list[UserDto]:
        """分页查询"""
        pass
